### DISCUSSION TITLE
We need you to grow Astroid together!

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#299] [Announcements]
Hi all,

Thank you for your interest in the Astroid project. I'm Sonny from TemPlaza. I've been a fan of Astroid ever since Naveen(JoomDev) introduced me to it. Unfortunately this project was later discontinued by JoomDev. I then stood up to continue to upgrade Astroid as a non-profit project under the GNU/GPL license.

Fortunately, with me, I have a few friends like @Chacapamac  and Peter (@Joomlaplates) who help me to support Astroid. Now I want to continue to develop Astroid but there is too much work to do alone in fact. So I need your help to move forward with Astroid. I need volunteers in the following areas:

1. Rewrite the documentation system and update with the latest Astroid version.
2. Make video tutorials so that users can easily access Astroid visually.
3. Manage the astroidframe.work site and write articles on the blog site.

If you can join with us, please tell me and we can go together :)
Please join our Discord channel here https://discord.gg/2MwtsAX4Py

PS: @Chacapamac @Joomlaplates May you help me update Astroid documentation? If yes, I will add you to GitBook and we can start a new documentation for Astroid asap

Thanks & Best Regards,
Sonny

**Comment by Chacapamac** [comment]
I @sonvnn , 
What is Gitbook and how its related with the Documentation on the Astroid Website?
Good Day!

**Comment by Chacapamac** [comment]
Also @sonvnn  can you continue with at least the Changelog of each version in the website blog. The new features for each can be the source of blog articles to give details how to use the features and after copy to the documentation.

I will try my best to help a bit more. Middle of next Year I will have more time for Astroid, hopefully.

**Comment by sonvnn** [comment]
Hi @Chacapamac 
GitBook is online documentation platform by GitHub. You can easy create online documentation in few minutes. It's free for personal.
I created a workplace for Astroid on GitBook here https://docs.astroidframe.work/ I invited you to join this project. Please take a look on it and let me know if you have any question.

PS: I am still working on Blog https://astroidframe.work/blog

Thanks for your support my friend :)
Best Regards,
Sonny

**Reply by Chacapamac** [reply]
**How do you integrate this in the main Astroid website ???** —> https://astroidframe.work/

**You want to centralized the infos at one place and the main website should be that place.** 
You don’t want people jumping to Gitbook, Github, Youtube... to have different piece of information about Astroid, this is not a good UI and confuse and discourage Astroid users.

**What you made with Joomla at https://astroidframe.work/support/documentations is perfect. I’m sure we can go further to render this even better.**

If you look at the competition _(Gantry, Helix, T3, YOOtheme, Wright Framework) Gantry, Helix, T3, YOOtheme, Wright Framework...)_ they **ALL** do exactly that.
 If you need information, download, support or interact with them, you go to one centralized place, their website.

**What you think ?**

**Reply by sonvnn** [reply]
Hi,

> How do you integrate this in the main Astroid website ???

By create a Menu (External Url Type) we can integrate docs.astroidframe.work to original website. It's very easy.

UX of current documentation is not well at all. So I would like find another way to help user can research Astroid. GitBook is most popular documentation tool. Our team is using it for our template and it really great option for Astroid. I will create the first steps in GitBook and you can see how it works :)

Thanks & Best Regards,
Sonny

**Reply by Chacapamac** [reply]
Great, if you can integrate it in the website like it is part of the website, I’m in.
**• Do you have an example of such an integration, just to see?**
**• You can add me and I will start.**

**• Can documentation Articles can be on a standby mode?** _(not visible by audience but only administrators. Like that, others administrators can review articles to approve or give their input)_.

**Comment by joomlaplates-de** [comment]
Hello, sorry for late response I am currently very busy with migration from J3 to J4 on my own site.
Anyway I could build a nice documentation website in Joomla 4 for the Astroid Framework. After that we can put the docs to the org Domain https://astroidframe.work/ 
But finally the org Astroid site is not that bad. Maybe just a bit cooler design for better overview.
Greatings Peter

**Reply by sonvnn** [reply]
Thanks Peter!

I sent you an invite. Now you can go to astroid documentation and arrange it. Please let me know if you have any questions.

Thanks & Best regards,
Sonny

**Comment by paurray** [comment]
Hi @sonvnn 

I am a part time Joomla dinosaur.
Have a lot of very superficial Joomla knowledge.

One thing I do know is that Astroid is fantastic.
The community here is great & I am very grateful to people like @Chacapamac  

My day job is as a video editor including tutorials and many other things.
You can visit my (built with Astroid) site here: https://finalbug.tv

I work with a lot of clients who have a very vague idea of what they want to do.
Much of my time is spent making infinite versions until they are happy.
I just do not have the brain space to do that here.
Nothing to do with you guys just where I am at.

If there is a clear detailed step by script/concept/road plan I can help out with the tutorials.

As in I can:

- Make screencasts.
- Edit them together.
- Add intro/outro if provided eg Logo, motto, call to action, jingle etc.
- Add a decent sounding voice over in English (native speaker from Ireland)
- Prepare for You Tube so things look and sound decent.

Any way if there are 3 things, structure, structure and structure I am on board.

My proposal to help me help the Astroid community.

Prepare for me the following:

- Clear list of topics and in what order.
- Create step by slides/screen shots for each topic.
- Label and number slides/screen shots for topics with an explanatory description.

THEN

- I convert the above steps into videos with voice over!

All the best

Paul

**Reply by sonvnn** [reply]
Wow! That's wonderful! Thank you for your help. 
I am preparing documentation for latest Astroid with Joomla 4. I have a planning to push them to Astroid website in April 2023. I will make it in clear list of topics and create step by slide/screenshots for each topic. After that I hope you can follow us and help us create tutorials videos for Astroid.

Thank you very much!
Best Regards,
Sonny

**Comment by paurray** [comment]
Just my 2 cents...

I watched many Astroid tutorials online about 6 months ago.
Many focused on the amazing functionality of Astroid.
But I found very little about the nuts & bolts of getting content to show up on a site.

There was a tutorial for using the layout section in Astroid.
But the tutorial I found was just putting content on one page.
I got stuck when trying to get different content to show up on different pages.

Admittedly I am not a hard core Joomla user.
And in some ways coming from Page Builder (Joomshaper) did not help.
As in I was accustomed to having a page builder to help me set things up on a page by page basis.

I think going into some of the Grammar of building a site with Astroid would help.

Hope this makes some kind of sense.

best

Paul

**Reply by Chacapamac** [reply]
This is cool @paurray 

> But I found very little about the nuts & bolts of getting content to show up on a site.

Yes, you right, videos/documentation on specific functionalities are great but Videos on How to actually work and build a website with Astroid will be great.

Help is welcome, also most of the videos on the Astroid Youtube Channel are old and represent mostly Astroid when Joomdev was the developers. Lot of work to do there.
https://www.youtube.com/@astroidframework2932/videos

**Comment by paurray** [comment]
The more I think about this the more I think that just videos illustrating the user manual will not help much :-)

_**!!! I will update the below on & off until the 1st of April. Any concrete feedback above & beyond Emojis is very welcome ;-) !!!**_

********************************************

**Lesson 01 : Install and activate**

- Finding and installing the Astroid framework and Template package
- Navigating to the Astroid Templates
- Switching between the different default Templates

**Lesson 02: Out of the box functionality**

- Switch on show module position in Joomla
(https://docs.joomla.org/Module_Position)
- Navigate to the Astroid Template Editor back end and look at standard Astroid Module positions available
- Mention that these are things that can set up quickly and easily again and again
- Template Header
- Miscellaneous/Copyright/Branding
EG Powered by “Astroid” is in the miscellaneous section / Module Position: footer-1
- Miscellaneous/Contact Information/
EG 1826 Locust Street, Bainbridge, GA 31717 / Module Position: topbar-1
- Social/Social Profiles
EG Facebook, Twitter & Co / Module Position: topbar-2
- Swap out/ switch on/off some of the Astroid features  EG Contact/topbar-1 and Social/topbar-2 positions- Compare "Joomla 

!!! **TO BE CONTINUED HERE** !!!

**Lesson 03: Creating a bare bones single page site with almost zero formatting**

- Create a custom Joomla module
- Enter a text eg "Hello World"
- Select the "Astroid Content Bottom" module position
- Publish, save and check the front end
- Repeat for a 2nd and 3rd module eg "Contact" and "Hero" joomla custom content modules
- Publish save and check the front end
- Swap at the order that the modules are displayed in on the front end
http://blog.astemplates.com/change-order-modules-joomla-3/
- Navigate to the Astroid Layout Menu and show where the "Astroid Content Bottom" lives

!!! **FROM HERE ON STILL A MESS** !!!

**Lesson 04: Creating a multi page site with almost zero formatting**

- Select the Astroid Content Top section, edit it and label it something like single front page
- Look at the Edit Element option in a section in the Astroid Layout Builder and note that there are module positions 
- Check to see if/what is happening in the front end
- Mention that to create custom content for our Astroid Site we need to jump through some hoops.
- Have a brief look peak at the default Joomla "Editor" to be used for editing content in Custom Modules
- Show how to clear Astroid Cache but point out that there is nothing to clear so far

**Continuing: Inserting a very basic "Hello World" in: Joomla**

- Navigate back to the Joomla admin home dashboard 
- Create a Joomla Menu item or just use default home page
- Create a 2nd Joomla Menu item eg Hero (use to double check modules)
- Save, publish and make sure that new menu item is showing
- Create a custom “My First Module” using the Joomla Module Manager
- Make sure that the Module is assigned to the page created above/home page
- Create unique module names like “Place holder for Hello World Section”
- Note the standard module positions and create unique module positions 
- Create a module unique position eg “Hello World Section”
- Make sure that the Module is published
- Add the text “Hello World”
- Check to see if/what is happening in the front end

**Joining the dots: Show "Hello World" text in Joomla Custom Module using Astroid Layout Builder**

- Navigate back to the Astroid Template you are working in
- Go to the Layout section and the “Home Page/Hello world”
- Go to the Edit Element Tab for this Layout Section
- Set the Module position to the “Hello World Section”
- Make sure that you have saved
- Check to see if/what is happening in the front end
- Clear Cache Function if necessary…

!!! Work in Progress here/Need to check out exactly how this works…

**Formatting text in Joomla Editor versus Astroid Color/Typography section**

- In Astroid create a copy of “Hello World” section
- Check that the copy is displaying in the front end
- In Joomla format the copy of “Hello World” section using the standard Joomla editor
- Change the font colour and font type in the copy of “Hello World” section
- Check changes displaying in the front end: “Hello World Section” 
- Back in the Astroid Color Section change Text Body Color
- Observer changes in front end
- Back in the Astroid Typography Section change Text Body Font Type
- Observer changes in front end
- Clear Cache Function if necessary…

…Need to check out exactly how this works. Work in Progress here…!!!

**Formatting layout out of multiple Modules on one page in: Joomla and Astroid**

- Set up/Edit Column/Multiple Columns on a Joomla page
!!!This is where things start getting hazy for me!!!
- Formatting different text sizes eg h1, h2 etc
- Check out the _Typography Tab_ in the main Astroid menu
- Make changes to the text formatting and check in the front end
- **_!!! Work in Progress!!!_**
- Show of some of the _Basic Astroid_ setting here eg; Pre Loader, Favicon, Back to Top etc

**Working with multiple Modules on multiple Site pages in: Joomla and Astroid**

- Set up multiple Joomla pages, eg in addition to Home & Hero also Contact
- Getting specific Modules to show on specific pages
- **_!!! Work in Progress!!!_**
- Show of some of the _Astroid Menu_ options at this point...
- **_!!! Work in Progress!!!_**

********************************************

**_Enough for now…but much of what is in the documentation…
... again concrete feedback above & beyond Emojis is very welcome ;-)_**

**Reply by sonvnn** [reply]
Hello @paurray
Please join our Discord channel https://discord.gg/2MwtsAX4Py
We need your help with Video Tutorials

**Comment by sonvnn** [comment]
Hi @Chacapamac @paurray @Joomlaplates 

I moved all documentation of Astroid to GitBook. Now, you can access to new documentation system via Original site https://astroidframe.work/ or https://docs.astroidframe.work/

Please give me some feedback about it.

Thanks & Best Regards,
Sonny

**Comment by pkEwando** [comment]
Hi Sonny, wow it looks really good and much better than before.

**Reply by sonvnn** [reply]
Thank you :)

**Comment by Chacapamac** [comment]
@sonvnn 
It look good and I’m difficult to please. Integration with the Astroid Website/Support/Download is nice.

**— Ready to help, will need some access....**

**Just find this small bug**

![Screen Shot 2023-03-20 at 9 45 13 AM](https://user-images.githubusercontent.com/1907276/226359437-a80bda3d-7cac-4aff-9983-bea25b927730.jpg)

**Comment by paurray** [comment]
@sonvnn Looks amazing.

I need to take a detailed look at it…

Off the top of my head: Suggestion 01/Installation Section

At the end of the installation section add something like this:

- If you have installed the complete Astroid Framework Template you can click on the “Get Started” button
- In the Joomla Templates Styles section click the radio button “Set as Default” for an Astroid Template of choice
- Click on the name of the chosen Astroid Template that is now set as default
- Then click on “Template Options” button in the details section to start editing the Astroid Template
- Here you can configure the Astroid Template and control the Layout of the Template
- You might want to consider making a copy of the Template before making changes

**Reply by sonvnn** [reply]
Thank you. If you would like to improve this documentation. Please give me your email, I will send you an invite to edit documentation.

Best Regards,
Sonny

**Reply by Chacapamac** [reply]
p.toulze@grafcomm.ca
**We need to find a way to know who’s working on what and have some kind of system of approbation?**

**Reply by sonvnn** [reply]
Invite sent. Don't worrry about it. This system works like Github. So we can track/edit/commit/merge edition very easy. You can learn more about it here https://docs.gitbook.com/

**Reply by paurray** [reply]
I can be reached here: paul(AT)finalbug.tv please send an invite.
I think the documentation is great. Really.
Just think that a little bit of cement to help stick all the building blocks together would go a long way :-)
Will take some time (10ish days) at my end but will happen.

**Comment by joomlaplates-de** [comment]
Hi Sonny, the documentation looks perfect and much better than before.
But how we can organize who will edit or update the docs?
Greatings Peter

**Reply by sonvnn** [reply]
It works like github. Author can edit/commit/merge their edition.

**Reply by sonvnn** [reply]
You can learn about it here https://docs.gitbook.com/

**Comment by Chacapamac** [comment]
Cool, Ok, I will take some time to look at Gitbook Docs and Come Back.

Personally, the first thing to do is to see if the actual Astroid Gitbook is user friendly in term of interface. I mean by that, if the way it’s structured now will really help the users. I will also take a look and come back...

**Comment by Chacapamac** [comment]
Ready when you are @sonvnn  — You can send me the login at my email — I will not change anything until review here first.

**Reply by sonvnn** [reply]
I sent you an invite email to your mailbox. Please help me check it.

Best Regards,
Sonny

**Reply by Chacapamac** [reply]
Just look and I cannot see the invite, I check my spam box and on my mail server....

Can you try another time and tell me the time you sent it?

Good Day @sonvnn !

**Reply by sonvnn** [reply]
Please use this link to join our project. https://app.gitbook.com/invite/5sakhWedcsRhvQ9ocEdo/vYo02zUZfOLvNDh2hPPa
Please let me know if you have any issue with it.

Best Regards,
Sonny

**Reply by Chacapamac** [reply]
**Tank You @sonvnn , it work!**

Curiously, first, it didn't let me login with my Github, account saying that “**account exist but with different credential**”
— I enter with Google:
— Come back soon after I explore a bit, Great!

**Reply by Chacapamac** [reply]
**Two first Questions @sonvnn** 

• When I press the “**Edit**” button
![Screen Shot 2023-04-18 at 2 17 04 PM](https://user-images.githubusercontent.com/1907276/232868937-6b81d26e-9ff4-4d57-aae5-d3e05ad4540c.jpg)
— A “**Draft**” is created, how can I delete them if I don’t want them?
— I guess each “**Draft**” show whatever changes you want to apply to a Section/Page and only an Administrator _(You)_ can accept/merge/commit the Draft ?

**Comment by Chacapamac** [comment]
First Question @sonvnn 

Each time I press ![Screen Shot 2023-04-18 at 2 17 04 PM](https://user-images.githubusercontent.com/1907276/232867659-93db32db-63e2-4861-b0ec-fa80aab2da0e.jpg)

**Comment by Chacapamac** [comment]
**Two First questions @sonvnn !**

• **Each time I press** ![Screen Shot 2023-04-18 at 2 17 04 PM](https://user-images.githubusercontent.com/1907276/232867910-0716392a-32d0-4a34-841d-c55de3d87e8c.jpg)
— A new Draft is created, how can I delete them if I don’t want them ?
— I guess draft show any change you want to bring to a page and an administrator (you) can accept it?

**Reply by sonvnn** [reply]
@Chacapamac I upgraded your permission to Creator role. So you can create, publish, merge and edit, right now.
![Screen Shot 2023-04-19 at 09 56 59](https://user-images.githubusercontent.com/3204103/232955272-103496a8-eecc-4c00-b3a2-caf5202aaae0.png)

**Reply by Chacapamac** [reply]
Great! Thank You @sonvnn !

**Comment by joomlaplates-de** [comment]
Hi all,
we need a kind of system to communicate who is editing anything on documentation site.
Otherwise it becomes a chaos. Maybe open a new channel with limited moderators.
Any ideas?

**Reply by sonvnn** [reply]
@Joomlaplates @Chacapamac Please join Discord channel to discuss more detail about Astroid future https://discord.gg/2MwtsAX4Py

Best Regards,
Sonny

**Reply by Chacapamac** [reply]
@Joomlaplates 
> Hi all, we need a kind of system to communicate who is editing anything on documentation site. Otherwise it becomes a chaos. Maybe open a new channel with limited moderators. Any ideas?

I just connect with Sonny on **Discord** _(see above link from Sonny)_
So far I really like the tool and I was worry about the same thing and ask Sonny about that on **Discord** in the **Documentation** section.

If I understand well Gitbook and reaffirm by Sonny this is how we can work and change the documentation pages.

**Process to modify EXISTING documentation on Gitbook**
— Select a Menu/Page — e,g. “Preloader”
— Press “**Edit**” blue Button or “**Change Request**”
— This create a **Draft** where you can do some changes to the document without altering the original _(**Draft**)_
— **Anybody can see all drafts in process by all others participants and comments  on each others**
— When people agree that a draft is finish, It can be pass to a final “Review” state and finally merged  by Administrators/Creators to the original public document accessible at —> https://docs.astroidframe.work/

**NOTES:**
— Administrator & Creators can merge without waiting for review.
— Administrator & Creators have permission to check "Change Requests" and merge it too.

Not sure yet how to add or remove pages and all the other details that go with the edition but it seem straightforward (after a while...)

**Reply by sonvnn** [reply]
@Chacapamac You are correct. 
PS: Only Admin and Creator can create/publish/edit/remove pages.

**Reply by sonvnn** [reply]
@Joomlaplates I did not see you in Gitbook yet. When you are ready, please let me know. We need your hand to improve our documentation. I saw your documentation in Joomlaplate.com. They are amazing :)

Best Regards,
Sonny
